tango_simlib/tango_sim_generator.py

```diff
@@ -281,13 +283,27 @@ def initialize_dynamic_attributes(self):
                 # Dynamically add all attributes except those with DevEnum data type
                 # and SPECTRUM data format since they are added statically to the
                 # device class prior to start-up.
-                if str(attr_dtype) != 'DevEnum' and str(d_format) != 'SPECTRUM':
+                if str(attr_dtype) == 'DevEnum':
+                    continue
+                elif str(d_format) in ['SPECTRUM', 'IMAGE']:
+                    continue
+                else:
                     # The return value of rwType is a string and it is required as a
                     # PyTango data type when passed to the Attr function.
                     # e.g. 'READ' -> PyTango.AttrWriteType.READ
                     rw_type = meta_data['writable']
                     rw_type = getattr(AttrWriteType, rw_type)
-                    attr = Attr(attribute_name, attr_dtype, rw_type)
+                    # Add a try/except clause when creating an instance of Attr class
+                    # as PyTango may raise an error when things go wrong.
+                    try:
+                        attr = Attr(attribute_name, attr_dtype, rw_type)
+                    except Exception as e:
+                        self._not_added_attributes.append(attribute_name)
+                        self._not_added_attributes_num += 1
```

Comment:
Why do you need a separate variable to keep track of the number of items in the list - can't you just use `len(self._not_added_attributes)` when you need the number?